### PR TITLE
1286: Fix bcast issue introduced with #883

### LINKED ISSUE #1286: Fix bug in broadcast introduced by #883
**Describe the bug**
```
The following tests FAILED:
         77 - vt_example:rdma_simple_put_4 (Failed)
         97 - vt:*/TestActiveBroadcastPut.test_type_safe_active_fn_bcast2/*_proc_4 (Failed)
         98 - vt:*/TestActiveBroadcast.test_type_safe_active_fn_bcast2/*_proc_4 (Failed)
        106 - vt:*/TestActiveBroadcastPut.test_type_safe_active_fn_bcast2/*_proc_8 (Failed)
        107 - vt:*/TestActiveBroadcast.test_type_safe_active_fn_bcast2/*_proc_8 (Failed)
        349 - vt:TestMemoryLifetime.test_active_bcast_serial_lifetime_proc_4 (Failed)
        351 - vt:TestMemoryLifetime.test_active_bcast_normal_lifetime_msgptr_proc_4 (Failed)
        357 - vt:TestMemoryLifetime.test_active_bcast_serial_lifetime_proc_8 (Failed)
        359 - vt:TestMemoryLifetime.test_active_bcast_normal_lifetime_msgptr_proc_8 (Failed)
        375 - vt:TestObjGroup.test_proxy_schedule_proc_4 (Failed)
        376 - vt:TestObjGroup.test_proxy_callbacks_proc_4 (Failed)
        382 - vt:TestObjGroup.test_proxy_schedule_proc_8 (Failed)
        383 - vt:TestObjGroup.test_proxy_callbacks_proc_8 (Failed)
        387 - vt:TestPhaseManagement.test_phase_manager_1_proc_4 (Failed)
        388 - vt:TestPhaseManagement.test_phase_manager_1_proc_8 (Failed)
        406 - vt:TestCallbackBcast.test_callback_bcast_1_proc_4 (Failed)
        407 - vt:TestCallbackBcast.test_callback_bcast_2_proc_4 (Failed)
        408 - vt:TestCallbackBcast.test_callback_bcast_3_proc_4 (Failed)
        409 - vt:TestCallbackBcast.test_callback_bcast_remote_1_proc_4 (Failed)
        410 - vt:TestCallbackBcast.test_callback_bcast_remote_2_proc_4 (Failed)
        411 - vt:TestCallbackBcast.test_callback_bcast_remote_3_proc_4 (Failed)
        423 - vt:TestCallbackBcast.test_callback_bcast_1_proc_8 (Failed)
        424 - vt:TestCallbackBcast.test_callback_bcast_2_proc_8 (Failed)
        425 - vt:TestCallbackBcast.test_callback_bcast_3_proc_8 (Failed)
        426 - vt:TestCallbackBcast.test_callback_bcast_remote_1_proc_8 (Failed)
        427 - vt:TestCallbackBcast.test_callback_bcast_remote_2_proc_8 (Failed)
        428 - vt:TestCallbackBcast.test_callback_bcast_remote_3_proc_8 (Failed)
        610 - vt:TestHighLevelCalls.test_rooted_lambda_proc_4 (Failed)
        611 - vt:TestHighLevelCalls.test_rooted_std_function_proc_4 (Failed)
        615 - vt:TestHighLevelCalls.test_rooted_lambda_proc_8 (Failed)
        616 - vt:TestHighLevelCalls.test_rooted_std_function_proc_8 (Failed)
        734 - vt:TestTermCleanup.test_termination_cleanup_2_proc_4 (Failed)
        735 - vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_4 (Failed)
        736 - vt:*/TestTermDepSendChain.test_term_dep_send_chain_merge/*_proc_4 (Failed)
        749 - vt:TestTermCleanup.test_termination_cleanup_2_proc_8 (Failed)
        750 - vt:*/TestTermDepSendChain.test_term_dep_send_chain/*_proc_8 (Failed)
        751 - vt:*/TestTermDepSendChain.test_term_dep_send_chain_merge/*_proc_8 (Failed)
```

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #1286

### CODE REVIEW

File: tests/unit/location/test_location.cc [hunk 1]
```diff
@@ -404,11 +404,12 @@ TYPED_TEST_P(TestLocationRoute, test_entity_cache_migrated_entity) /* NOLINT */{
     auto const nb_rounds = 5;
     auto nb_received     = 0;
 
-    // register entity
-    if (my_node == home) {
-      vt::theLocMan()->virtual_loc->registerEntity(entity, my_node);
-    }
-    vt::theCollective()->barrier();
+    ::vt::runInEpochCollective([my_node, entity] {
```
Comment (cz4rs) [inline tests/unit/location/test_location.cc:407]: should `home` be captured as well?

Comment (JacobDomagala) [reply tests/unit/location/test_location.cc:407]: Clang will issue a warning for this one:
https://godbolt.org/z/T6z65G

It's possible for lambda to use some values without the explicit capture.
From [cppreference](https://en.cppreference.com/w/cpp/language/lambda):
image

**Review by cz4rs — APPROVED** [review]
looks good to me

**Review by jstrzebonski — APPROVED** [review]
Seems to be OK

**Review by lifflander — APPROVED** [review]

**Pull request merged**